pkg/executor/internal/mpp/local_mpp_coordinator.go

```diff
@@ -336,14 +339,18 @@ func (c *localMppCoordinator) fixTaskForCTEStorageAndReader(exec *tipb.Executor,
 }
 
 // DFS to check if plan needs report execution summary through ReportMPPTaskStatus mpp service
-// Currently, return true if plan contains limit operator
-func needReportExecutionSummary(plan base.PhysicalPlan) bool {
+// Currently, return true if there is a limit operator in the path from current TableReader to root
+func needReportExecutionSummary(plan base.PhysicalPlan, destTablePlanID int, foundLimit bool) bool {
 	switch x := plan.(type) {
 	case *plannercore.PhysicalLimit:
-		return true
+		return needReportExecutionSummary(x.Children()[0], destTablePlanID, true)
+	case *plannercore.PhysicalTableReader:
+		if foundLimit {
+			return x.GetTablePlan().ID() == destTablePlanID
+		}
 	default:
 		for _, child := range x.Children() {
```

Comment:
I'm not sure if CTE also works. Its children will not be returned by calling `Children()`

Reply:
OK, I'll check it.